src/vm/eetwain.cpp

```diff
@@ -3099,7 +3099,17 @@ size_t EECodeManager::GetCallerSp( PREGDISPLAY  pRD )
     {
         EnsureCallerContextIsValid(pRD, NULL);
     }
-    return (size_t) (GetSP(pRD->pCallerContext));
+
+    size_t res = GetSP(pRD->pCallerContext);
+
+#ifdef _TARGET_X86_
+    EECodeInfo codeInfo;
+    codeInfo.Init((PCODE) pRD->ControlPC);
+
+    res -= codeInfo.GetCodeManager()->GetStackParameterSize(&codeInfo);
```

Comment:
@janvorli Oh, I'm sorry, I misunderstood your comment. It will work if your suggestion was to add ``DWORD CallerSp`` field in ``REGDISPLAY``. I thought that your suggestion was to update Esp as Caller SP.